docs/language-reference/07-declarations.md

```diff
@@ -0,0 +1,770 @@
+
+```hlsl
+struct Person
+{
+    var age : int;
+    float height;
+
+    int getAge() { return age; }
+    func getHeight() -> float { return this.height; }
+    static func getPopulation() -> int { ... }
+}
+```
+
+The body of a structure type declaration may include variable, type, function, and initializer declarations.
+
+### Fields
+
+Variable declarations in the body of a structure type declaration are also referred to as _fields_.
+
+A field that is marked `static` is shared between all instances of the type, and is semantically like a global variable marked `static`.
+
+A non-`static` field is also called an _instance field_.
+
+### Methods
+
+Function declarations in the body of a structure type declaration are also referred to as _methods_.
+
+A method declaration may be marked `static`.
+A `static` method must be invoked on the type itself (e.g., `Person.getPopulation()`).
+
+A non-`static` method is also referred to as an _instance method_.
+Instance methods must be invoked on an instance of the type (e.g., `somePerson.getAge()`).
+The body of an instance method has access to an implicit `this` parameter which refers to the instance on which the method was invoked.
+
+By default the `this` parameter of an instance method acts as an immutable variable.
+An instance method with the `[mutating]` attribute receives a mutable `this` parameter, and can only be invoked on a mutable value of the structure type.
+
+### Inheritance
+
+A structure type declaration may include an _inheritance clause_ that consists of a colon (`:`) followed by a comma-separated list of types that the structure type inherits from:
+
+```
+struct Person : IHasAge, IHasName
+{ .... }
+```
+
+When a structure type declares that it inherits from an interface, the programmer asserts that the structure type implements the required members of the interface.
+
+### Syntax Details
+
+A structure declaration does *not* need to be terminated with a semicolon:
+
+```hlsl
+// A terminating semicolon is allowed
+struct Stuff { ... };
+
+// The semicolon is not required
+struct Things { ... }
+```
+
+When a structure declarations ends without a semicolon, the closing curly brace (`}`) must be the last non-comment, non-whitespace token on its line.
+
+For compatiblity with C-style code, a structure type declaration may be used as the type specifier in a traditional-style variable declaration:
+
+```hlsl
+struct Association
+{
+    int from;
+    int to;
+} associations[] =
+{
+    { 1, 1 },
+    { 2, 4 },
+    { 3, 9 },
+};
+```
+
+If a structure type declaration will be used as part of a variable declaration, then the next token of the variable declaration must appear on the same line as the closing curly brace (`}`) of the structure type declaration.
+The whole variable declaration must be terminated with a semicolon (`;`) as normal.
+
+
+Enumeration Types
+-----------------
+
+Enumeration type declarations are introduced with the `enum` keyword:
```

Comment:
Missing `[UnscopedEnum]` `[Flag]`, underlying type, builtin operator overload for enums (& |).